third_party/ruby/build-ruby.bzl

```diff
@@ -280,13 +287,16 @@ def _build_ruby_impl(ctx):
             hdrs = " ".join(hdrs),
             libs = " ".join(libs),
             rubygems = ctx.files.rubygems[0].path,
+            libyaml = ctx.files.libyaml[0].dirname,
+            rustc = ctx.toolchains["@rules_rust//rust:toolchain_type"].rustc.path,
             configure_flags = " ".join(ctx.attr.configure_flags),
             sysroot_flag = ctx.attr.sysroot_flag,
             install_extra_srcs = "\n".join(install_extra_srcs),
             extra_srcs_object_files = " ".join(extra_srcs_object_files),
             install_append_srcs = "\n".join(install_append_srcs),
             install_gems = "\n".join(install_gems),
         )),
+        tools = [ctx.toolchains["@rules_rust//rust:toolchain_type"].rustc],
```

Comment:
At the moment this makes `rules_rust` a dependency for all ruby builds, we could make this selective instead and opt-out all pre-3.2 ruby versions and have it default on for 3.2 onwards.

Reply:
I tried to do this but unfortunately I couldn't make including this toolchain configurable. It'll be downloaded for all ruby builds but a no-op for anything < 3.2